storio/src/androidTest/java/com/pushtorefresh/storio/db/integration_test/impl/QueryTest.java

```diff
@@ -17,28 +19,170 @@
 public class QueryTest extends BaseTest {
 
     @Test public void queryAll() {
+        final List<User> users = putUsers(3);
+
+        final List<User> usersFromQuery = storIODb
+                .get()
+                .listOfObjects(User.class)
+                .withMapFunc(User.MAP_FROM_CURSOR)
+                .withQuery(new Query.Builder()
+                        .table(User.TABLE)
+                        .build())
+                .prepare()
+                .executeAsBlocking();
+
+        assertTrue(users.equals(usersFromQuery));
+    }
+
+    @Test public void queryOneByField() {
+        final List<User> users = putUsers(3);
+
+        for (User user : users) {
+            final List<User> usersFromQuery = storIODb
+                    .get()
+                    .listOfObjects(User.class)
+                    .withMapFunc(User.MAP_FROM_CURSOR)
+                    .withQuery(new Query.Builder()
+                            .table(User.TABLE)
+                            .where(User.COLUMN_EMAIL + "=?")
+                            .whereArgs(user.getEmail())
+                            .build())
+                    .prepare()
+                    .executeAsBlocking();
+
+            assertEquals(usersFromQuery.size(), 1);
+            assertEquals(usersFromQuery.get(0), user);
+        }
+    }
+
+    @Test public void queryOrdered() {
         final List<User> users = TestFactory.newUsers(3);
+        Collections.reverse(users);
+        putUsers(users);
 
-        final PutCollectionResult<User> putResult = storIODb
-                .put()
-                .objects(users)
-                .withMapFunc(User.MAP_TO_CONTENT_VALUES)
-                .withPutResolver(User.PUT_RESOLVER)
+        final List<User> usersFromQueryOrdered = storIODb
+                .get()
+                .listOfObjects(User.class)
+                .withMapFunc(User.MAP_FROM_CURSOR)
+                .withQuery(new Query.Builder()
+                        .table(User.TABLE)
+                        .orderBy(User.COLUMN_EMAIL)
+                        .build())
                 .prepare()
                 .executeAsBlocking();
 
-        assertEquals(users.size(), putResult.numberOfInserts());
+        Collections.sort(users);
```

Comment:
Please add a comment here `Users can be sorted by email`, I was quite confused when read it, but then I found that `User` now implements `Comparable`, not very obvious :)